core/jvmMain/src/kotlinx/serialization/internal/Platform.kt

```diff
@@ -153,6 +153,9 @@ private fun <T : Any> Class<T>.createEnumSerializer(): KSerializer<T> {
 }
 
 private fun <T : Any> Class<T>.findObjectSerializer(): KSerializer<T>? {
+    // Special case to avoid IllegalAccessException on Java11+
+    // There are no serializable objects in the stdlib anyway.
+    if (this.canonicalName.let { it.startsWith("java.") || it.startsWith("kotlin.") }) return null
```

Comment:
Rather than check for particular packages, the problem in the underlying bug is actually that the type itself is private. So what about just using `Modifier.isPublic(modifiers)` rather than a textual check?

Reply:
@sandwwraith Package private would still not be correct on "public", but I can see where you are over constraining the context where this function is called from within the package.